sentry-android-core/src/main/java/io/sentry/android/core/util/MainThreadChecker.java

```diff
@@ -0,0 +1,46 @@
+package io.sentry.android.core.util;
+
+import android.os.Looper;
+import io.sentry.core.protocol.SentryThread;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+
+/** Class that checks if a given thread is the Android Main/UI thread */
+@ApiStatus.Internal
+public final class MainThreadChecker {
+
+  private MainThreadChecker() {}
+
+  /**
+   * Checks if a given thread is the Android Main/UI thread
+   *
+   * @param thread the Thread
+   * @return true if it is the main thread or false otherwise
+   */
+  public static boolean isMainThread(final @NotNull Thread thread) {
+    return isMainThread(thread.getId());
+  }
+
+  /**
+   * Checks if the calling/current thread is the Android Main/UI thread
+   *
+   * @return true if it is the main thread or false otherwise
+   */
+  public static boolean isMainThread() {
+    return isMainThread(Thread.currentThread());
+  }
+
+  /**
+   * Checks if a given thread is the Android Main/UI thread
+   *
+   * @param sentryThread the SentryThread
+   * @return true if it is the main thread or false otherwise
+   */
+  public static boolean isMainThread(final @NotNull SentryThread sentryThread) {
```

Comment:
`m`: If I am not mistaken, we don't have a test for this.

Reply:
yep, I will add it